tests/trainer/test_trainer.py

```diff
@@ -1332,15 +1332,17 @@ def test_num_sanity_val_steps_neg_one(tmpdir, limit_val_batches):
         ),
     ],
 )
+# Todo: mock nb Gpus so all these tests can run on any device
+# todo: think about simplification, that the the expected will be just a list use_xxx which shall be true...
```

Comment:
This should fix these TODOs:

```python
@pytest.mark.parametrize(
    "trainer_kwargs,expected",
    [
        (
            dict(distributed_backend=None, gpus=None),
            dict(
                use_dp=False,
                use_ddp=False,
                use_ddp2=False,
                num_gpus=0,
                on_gpu=False,
                use_single_gpu=False,
                num_processes=1,
            ),
        ),
        (
            dict(distributed_backend="dp", gpus=None),
            dict(
                use_dp=False,
                use_ddp=False,
                use_ddp2=False,
                num_gpus=0,
                on_gpu=False,
                use_single_gpu=False,
                num_processes=1,
            ),
        ),
        (
            dict(distributed_backend="dp", gpus=None),
            dict(
                use_dp=False,
                use_ddp=False,
                use_ddp2=False,
                num_gpus=0,
                on_gpu=False,
                use_single_gpu=False,
                num_processes=1,
            ),
        ),
        (
            dict(distributed_backend="ddp", gpus=None),
            dict(
                use_dp=False,
                use_ddp=False,
                use_ddp2=False,
                num_gpus=0,
                on_gpu=False,
                use_single_gpu=False,
                num_processes=1,
            ),
        ),
        (
            dict(distributed_backend="ddp", num_processes=2, gpus=None),
            dict(
                use_dp=False,
                use_ddp=True,
                use_ddp2=False,
                num_gpus=0,
                on_gpu=False,
                use_single_gpu=False,
                num_processes=2,
            ),
        ),
        (
            dict(distributed_backend="ddp", num_nodes=2, gpus=None),
            dict(
                use_dp=False,
                use_ddp=True,
                use_ddp2=False,
                num_gpus=0,
                on_gpu=False,
                use_single_gpu=False,
                num_processes=1,
            ),
        ),
        (
            dict(distributed_backend="ddp_cpu", num_processes=2, gpus=None),
            dict(
                use_dp=False,
                use_ddp=True,
                use_ddp2=False,
                num_gpus=0,
                on_gpu=False,
                use_single_gpu=False,
                num_processes=2,
            ),
        ),
        (
            dict(distributed_backend="ddp2", gpus=None),
            dict(
                use_dp=False,
                use_ddp=False,
                use_ddp2=False,
                num_gpus=0,
                on_gpu=False,
                use_single_gpu=False,
                num_processes=1,
            ),
        ),
        (
            dict(distributed_backend=None, gpus=1),
            dict(
                use_dp=False,
                use_ddp=False,
                use_ddp2=False,
                num_gpus=1,
                on_gpu=True,
                use_single_gpu=True,
                num_processes=1,
            ),
        ),
        (
            dict(distributed_backend="dp", gpus=1),
            dict(
                use_dp=True,
                use_ddp=False,
                use_ddp2=False,
                num_gpus=1,
                on_gpu=True,
                use_single_gpu=True,
                num_processes=1,
            ),
        ),
        (
            dict(distributed_backend="ddp", gpus=1),
            dict(
                use_dp=False,
                use_ddp=True,
                use_ddp2=False,
                num_gpus=1,
                on_gpu=True,
                use_single_gpu=True,
                num_processes=1,
            ),
        ),
        (
            dict(distributed_backend="ddp_cpu", num_processes=2, gpus=1),
            dict(
                use_dp=False,
                use_ddp=True,
                use_ddp2=False,
                num_gpus=0,
                on_gpu=False,
                use_single_gpu=False,
                num_processes=2,
            ),
        ),
        (
            dict(distributed_backend="ddp2", gpus=1),
            dict(
                use_dp=False,
                use_ddp=False,
                use_ddp2=True,
                num_gpus=1,
                on_gpu=True,
                use_single_gpu=False,
                num_processes=1,
            ),
        ),
        (
            dict(distributed_backend=None, gpus=2),
            dict(
                use_dp=False,
                use_ddp=True,
                use_ddp2=False,
                num_gpus=2,
                on_gpu=True,
                use_single_gpu=False,
                num_processes=2,
            ),
        ),
        (
            dict(distributed_backend="dp", gpus=2),
            dict(
                use_dp=True,
                use_ddp=False,
                use_ddp2=False,
                num_gpus=2,
                on_gpu=True,
                use_single_gpu=False,
                num_processes=1,
            ),
        ),
        (
            dict(distributed_backend="ddp", gpus=2),
            dict(
                use_dp=False,
                use_ddp=True,
                use_ddp2=False,
                num_gpus=2,
                on_gpu=True,
                use_single_gpu=False,
                num_processes=2,
            ),
        ),
        (
            dict(distributed_backend="ddp2", gpus=2),
            dict(
                use_dp=False,
                use_ddp=False,
                use_ddp2=True,
                num_gpus=2,
                on_gpu=True,
                use_single_gpu=False,
                num_processes=1,
            ),
        ),
    ],
)
def test_trainer_config(trainer_kwargs, expected, monkeypatch):
    if trainer_kwargs["gpus"] is not None:
        monkeypatch.setattr(torch.cuda, "is_available", lambda: True)
        monkeypatch.setattr(torch.cuda, "device_count", lambda: trainer_kwargs["gpus"])
    trainer = Trainer(**trainer_kwargs)
    for k, v in expected.items():
        assert getattr(trainer, k) == v, f"Failed {k}: {v}"
```

Reply:
nice! I would clean up the tests a bit more but for clarity that the functionality is not touched I would rather keep the tests as they are and you are welcome to send a PR just with this tests update :]

Reply:
Sure! I'll open a PR with this patch once this PR is merged